wordpress-mobile/WordPress-Android
# Introduce pagination for device library

This PR is solving a problem where we want to add pagination to the "Choose from device" list. This issue is actually quite complicated because we have 4 independent sources (all of which need to be paginated) and then we have to make sure we load the merged data in the correct order. The pagination is necessary because the content resolver mime type check is a heavy operation and if you have a lot of media files, it could show you an empty list for quite some time. 

Because the logic is complicated, I've added a lot of tests to cover all the possible scenarios (and extracted the android specific logic into a new file). Let me know what you think!

To test:
- Stress test the "Choose from device" loader
- try devices with many files/images, try that the pagination works as expected and nothing jumps around as you scroll
- try devices with no images
- check that loading isn't slow
- set the PAGE_SIZE to something smaller to make it easier to test pagination

PR submission checklist:
